### PR TITLE
Add patient notes card

### PR DESCRIPTION
**Pull request by atuonufure**
Added the ability to create and view patient notes in Markdown format,
allowing the practitioner to maintain a variety of patient notes

The notes card is a dashboard card on the patient's Overview page. Can be located anywhere on the Overview page.

This card can be used in any project using fhir-emr by adding the configuration to `dashboard.config.ts` as example

```typescript
{
    query: {
        resourceType: 'Observation',
        search: (patient: Patient) => ({
            subject: patient.id,
            code: '866145009',
            _sort: '-_lastUpdated',
        }),
    },
    widget: PatientNoteListCardContainer,
}
```

Note list:

image

Add note:

image

Open note:

image

### CODE REVIEW

File: src/components/BaseQuestionnaireResponseForm/widgets/MDEditorControl/MarkDownEditor/index.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                    return (
+                        <div className="MarkDownToolBar" style={{ display: 'flex' }}>
+                            <UndoRedo />
+                            <Separator />
+                            <BoldItalicUnderlineToggles />
+                            <CodeToggle />
+                            <Separator />
+                            <ListsToggle />
+                            {/* <Separator /> */}
+                            {/* TODO <CreateLink /> */}
```
Comment (vesnushka) [inline src/components/BaseQuestionnaireResponseForm/widgets/MDEditorControl/MarkDownEditor/index.tsx:61]: Please use TODO with readable description of what needs to be done and remove all the commented lines

File: resources/seeds/Questionnaire/patient-note-create.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    required: true
+    type: string
+    linkId: note-title
+  - text: Note
+    required: true
+    type: string
+    linkId: note-content
+    itemControl:
+      coding:
+        - code: markdown-content
```
Comment (vesnushka) [inline resources/seeds/Questionnaire/patient-note-create.yaml:29]: I would prefer to call the control `markdown-editor`

File: .eslintrc.cjs [hunk 3]
```diff
@@ -43,11 +43,11 @@ module.exports = {
         'react/react-in-jsx-scope': 'off',
         '@typescript-eslint/no-non-null-assertion': 'off',
         '@typescript-eslint/ban-ts-comment': 'off',
-        "string-to-lingui/t-call-in-function": 2,
+        'string-to-lingui/t-call-in-function': 2,
         'import/no-unresolved': [
             2,
             {
-                ignore: ['fhir/r4b', '@beda.software/emr-config', '@beda.software/aidbox-types'], // Fixes error: Unable to resolve path to module 'fhir/r4b'.
+                ignore: ['fhir/r4b', '@beda.software/emr-config', '@beda.software/aidbox-types', '@mdxeditor/editor'], // Fixes error: Unable to resolve path to module 'fhir/r4b'.
```
Comment (vesnushka) [inline .eslintrc.cjs:50]: What's the problem here? Doesn't @mdxeditor/editor include types?

**Comment by vesnushka** [comment]
yarn.lock changes looks huge ~1800 new lines, all of them related to @mdxeditor/editor?

File: .eslintrc.cjs (hunk 3 above)
Comment (atuonufure) [reply .eslintrc.cjs:50]: The types are there, but there are errors
`Unable to resolve path to module '@mdxeditor/editor'`
`Unable to resolve path to module '@mdxeditor/editor/style.css'`
Solution to this problem could not be found quickly

**Comment by atuonufure** [comment]
> yarn.lock changes looks huge ~1800 new lines, all of them related to @mdxeditor/editor?

Looks like it, I double-checked and used the previous yarn.lock and uninstalled node_modules, then installed the dependencies again - still ~1800 lines

**Pull request merged**